presets/README.md

```diff
@@ -16,3 +16,69 @@ Each preset model has its own hardware requirements in terms of GPU count and GP
 ## Distributed inference
 
 For models that support distributed inference, when the node count is larger than one, [torch distributed elastic](https://pytorch.org/docs/stable/distributed.elastic.html) is configured with master/worker pods running in multiple nodes and the service endpoint is the master pod.
+
+## OOM Troubleshooting
+
+### Gathering Debug Information
+1. Set CUDA Environment Variables:
+   - `CUDA_LAUNCH_BLOCKING=1`: This variable provides more detailed debug information
+   - Reduce GPU Memory Fragmentation by setting:
+      - `PYTORCH_CUDA_ALLOC_CONF=expandable_segments:True`
+      - `PYTORCH_CUDA_ALLOC_CONF=max_split_size_mb:512`
+
+
+## Tuning
+Debugging GPU OOM (Out of Memory) issues during the fine-tuning process involves several strategies. Here are some steps to help you resolve these issues:
+
+
+
+### Preliminary Steps
+If setting environment variables doesn't resolve the issue, consider these steps:
+1. Reduce Batch Size: 
+   - The default batch size is set to the smallest size of 1. If you haven't modified this, you are already using the minimum batch size.
+2. Enable QLoRa (Quantization)
+   - Refer to `../charts/kaito/templates/qlora-params.yaml` for an example. (TODO: example GRaph)
+3. Truncation and Padding
+   - Truncation reduces data sample sizes, which can help prevent OOM errors. For more information, refer to [Hugging Face's documentation on padding and truncation](https://huggingface.co/docs/transformers/en/pad_truncation).
+
+There are several ways to go about debugging GPU OOM issues during the fine-tuning process. 
+
+
+### Observing and Adjusting the Dataset
+
+If issues persist, examine your tuning dataset for outliers:
+- For example see this dataset below ![img.png](dataset_sizes.png)
+- Identify and filter out larger data samples that may be causing OOM issues.
+- Consider truncating data samples to manage their size.
+
+### Memory Management
+
```

Comment:
You definitely need to explain the reserved memory and allocated memory here and how they will affect the OOM.